src/plugins/mer/mersettings.cpp

```diff
@@ -198,6 +204,36 @@ void MerSettings::setAskBeforeClosingVmEnabled(bool enabled)
     emit s_instance->askBeforeClosingVmEnabledChanged(s_instance->m_askBeforeClosingVmEnabled);
 }
 
+FileName MerSettings::deviceModelFileName()
+{
+    QSettings *globalSettings = PluginManager::globalSettings();
+    return FileName::fromString(QFileInfo(globalSettings->fileName()).absolutePath()
+                                + QLatin1String(MER_DEVICE_MODELS_FILENAME));
+}
+
+MerEmulatorDeviceModel MerSettings::deviceModel(const QString &name)
+{
+    Q_ASSERT(s_instance);
+
+    return s_instance->m_deviceModels.value(name);
+}
+
+QStringList MerSettings::deviceModelNames()
+{
+    Q_ASSERT(s_instance);
+
+    QCollator collator(QLocale::system());
```

Comment:
This looks to heavyweight to me. There is no use of QCollator in Qt Creator code. I would simply rely on the default QString::operator< utilized when QString is used as a key to a QMap. I.e., if MerSettings::deviceModels() returns QMap<QString, MerEmulatorDeviceModel> as I suggested before, then deviceModels().keys() already provides list of names in the desired order.

Reply:
User will able to set profile name. The profile name may be in any locale and therefore simple < operator may not work properly (Custom1 and Custom11, for example). So I use QCollator and in next commit QMap changed to QHash.

Reply:
I must keep asking how come it works elsewhere in Qt Creator? Qt Creator is definitely being used under various locales and yet grepping for QCollator does not reveal single occurrence in the whole code base.

If you really think this is needed then please move it near the UI layer. I.e. do the sorting just before populating a view. Feel free to expose a helper function in the MerPlugin class as there is currently no better place suitable for exposing such a function.

Reply:
QCollator used in QFileSystemModel for proper sorting: [http://code.qt.io/cgit/qt/qtbase.git/tree/src/widgets/dialogs/qfilesystemmodel.cpp?#n1095](http://code.qt.io/cgit/qt/qtbase.git/tree/src/widgets/dialogs/qfilesystemmodel.cpp?#n1095). QString operator< sorting array like "New1", "New2", "New11" wrong, but QFileSystemModel using QCollator sort files with such names properly.

Reply:
I meant Qt Creator sources, not Qt sources :)

Reply:
I know. QtCreator has bug (or feature) with sorting: just create projects "New2", "New1", "New11" in this order in one session and you'll see that projects are sorted and that they are sorted wrong. Anyway, I moved QColator to nearer to GUI level.